src/lib/atoms/Avatar/Avatar.tsx

```diff
@@ -1,68 +1,118 @@
      */
-    shape?: 'circle' | 'square';
+    isLoading?: boolean;
     /**
-     * Handle click event on avatar component((event: Event) => void 0).
+     * Additional class for the parent element.
+     * This prop should be used to set placement properties for the element relative to its parent using BEM conventions.
      */
-    onClick?: (event: MouseEvent) => void;
+    className?: string;
 }
 
-const Avatar: React.FC<IAvatarProps> = ({ src, icon, size, color, shape, onClick, children }) => {
-    return (
-        <div
-            className={classnames('user-avatar-c', {
-                [`size-${size}`]: size,
-                [`color-${color}`]: color,
-                [`shape-${shape}`]: shape,
-                [icon!]: !src && icon,
-                'text-small': typeof children === 'string' && children.length > 1,
-                clickable: !!onClick
+const iconSizes = {
+    small: 16,
+    large: 20,
+    '6Xlarge': 48
+} as const;
+
+const AvatarWrapper = ({ onClick, children, parentClass }) => {
+    return onClick ? (
+        <button onClick={onClick} className={`${parentClass} avatar_button`}>
+            {children}{' '}
+        </button>
+    ) : (
+        <div className={parentClass}>{children}</div>
+    );
+};
+
+/**
+ * An avatar is a graphical representation of a user, typically displayed as a small image or icon. It can be a photo, illustration, or initials, and is used to personalize the user experience by visually identifying the user in interfaces such as profiles, comment sections, and messaging apps.
+ */
+const Avatar: FC<IAvatarProps> = ({
+    size = 'medium',
+    color = 'magenta',
+    fullName,
+    src,
+    onClick,
+    isDisabled,
+    isLoading,
+    Icon = <Square />,
+    className
+}) => {
+    const [cutFirstAndLastName, setCutFirstAndLastName] = useState<string>('');
+
+    useEffect(() => {
+        if (!fullName) {
+            setCutFirstAndLastName('');
+            return;
+        }
+        const [firstName, lastName] = fullName.split(' ');
+
+        setCutFirstAndLastName(
+            `${firstName[0]}${lastName && (size === '6Xlarge' || size === 'large') ? ' ' + lastName[0] : ''}`
+        );
+    }, [fullName, size]);
+
+    const iconMock =
+        Icon &&
+        cloneElement(Icon, {
+            size: iconSizes[size]
+        });
+
+    return isLoading ? (
+        <span>skeleton</span>
+    ) : (
+        <AvatarWrapper
+            parentClass={classNames(`avatar avatar_size_${size} avatar_color_${color}`, className, {
+                avatar_disabled: isDisabled
             })}
-            style={{ backgroundImage: src ? `url(${src})` : '' }}
             onClick={onClick}
         >
-            {!src && !icon && children && getInitials(children)}
-        </div>
+            {iconMock && !cutFirstAndLastName && iconMock}
+            {src && <img className="avatar__image" alt={'avatar'} src={src} />}
```

Comment:
Here is exist issue in rendering `src` and `icon` will be rendered together handle logic